src/rank_llm/scripts/run_rank_llm.py

```diff
@@ -153,5 +155,10 @@ def main(args):
         default="You are RankLLM, an intelligent assistant that can rank passages based on their relevancy to the query.",
         help="the system message used in prompts",
     )
+    parser.add_argument(
+        "--batched",
```

Comment:
This batch can be confused with rerank_batch, the latter is for reranking multiple queries regardless of if it is happening one query at a time or multiple queries at once. The former refers to reranking multiple queries at once. from the default param values it looks like we want to support rerank_batch function with batch=false? if yes, I think the two different meanings of batch can be confusing, maybe we need to rename one of the two usecases? WDYT @ronakice , @lintool

Reply:
Hmm i agree this could be a source of confusion for users. Batching usually means multiple queries being processed at once correct? In which case `rerank_batch` needs renaming?

Reply:
with my understanding batch processing is async and yes it is normally multiple queries at once, but it doesn't restrict the batch size to be strictly greater than 1. IIRC, @lintool  chose `rerank` and `rerank_batch` to align with pyserini function names (`retrieve` and `retrieve_batch`).
@ronakice WDYT about changing `--batched` to `--vllm-batched`?
 I propose to have an `--inference-method` enum (or some similar name) with FastChat as the default value. The user can set it to vllm if they want to use vllm rather than fastchat inference.

Reply:
I can add these two yes 👍🏼 I agree for the time being this is more clear

Reply:
Less than Ideal since batch in rerank_batch has a different implication than batch in vllm-batched, but it is ok for now